warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/WhindexFieldIngestHelper.java

```diff
@@ -0,0 +1,242 @@
+package datawave.ingest.data.config.ingest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.ImmutableMultimap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Multimap;
+
+import datawave.ingest.data.Type;
+import datawave.ingest.data.config.NormalizedContentInterface;
+import datawave.ingest.data.config.NormalizedFieldAndValue;
+
+/**
```

Comment:
Similar comment here about simplifying the class-level documentation. The internal properties shouldn't necessarily be described here since they already have documentation at their definition. You should include some documentation about how setup() must be called first to initialize the whindex configurations before processWhindexFields() can be used to generate the whindex entries.